VidLoader/VidLoader/Sources/VidLoader.swift

```diff
@@ -257,13 +257,20 @@ public final class VidLoader: VidLoadable {
     }
 
     private func startTask(urlAsset: AVURLAsset, streamResource: StreamResource, item: ItemInformation) {
+        addCookiesTo(urlAsset: urlAsset, headers: item.headers)
```

Comment:
Hi @e-nicolas do you have an URL that I can use to test?
I think you need to send `item.headers` in `setupResourceDelegate` to`ResourceLoader`, isn't AVAssetResourceLoaderDelegate `CookieLoader` replaced with `ResourceLoader` when `setupResourceDelegate` is called?

Reply:
Hi @Cyklet, thanks for your quick reply. I saw that the headers was not necessary to be added on URLRequest header fields. I just added the cookies on HTTPCookieStorage and pass it to AVURLAsset too. This way started to download de videos properly.

On VideoListDataProvider I added the headers:

`let headers = ["CloudFront-Key-Pair-Id": "K8IK9YH5HAVIE", "CloudFront-Policy": "eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9xYS1tZWRpYS5hZnlhLmNvbS5ici82MDA4MzRiNTY3MjYwZDIxOTBmM2Y0NTkvKiIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTY1MjQ3MjY1OH19fV19", "CloudFront-Signature": "at1Q5KAYsweDfq7W3jKT9uPK51ZGaBo~32MivGBLjFTO~ZPi6qsvI8XrtdEba0b-wOZX9OaRmPS1DU0ri7tCzWJYh1harwfGRLEbc2LlOnv7POsxySDCuSEnK0AFnP~cIuisCZa-sbsZGmTWJSVh~iJqXi5JljYK66P163OVCccQqC1Pkdw-QulrCCAEyfOUEkmCva8gljpSpspHaWxJPWsBPGWVoXqbLQ0vZAm9Ybe3HS-da1Vc6voLTfxMNgH2lWHiFVpvpKN-88QIUltsLcT9fHW~pPq7jduXE8wBgEaw4E-uGAEiH1wSCbHwzeHierVjjo6eTh1ZctrEIyY0rA__" ]`

And pass it to `DownloadValues` on `startDownload` method.

I changed the video defaultURL to:
https://qa-media.afya.com.br/600834b567260d2190f3f459/master.m3u8

This signature and policy has duration of 1 day